### ISSUE TITLE
Web UI - 500 Error with 2.3.6

### ISSUE THREAD

**Issue by Dot50Cal** [mrlt8/docker-wyze-bridge#885]
Hey all, I had to fall back a few versions to get things functional again. Error appears to be the same from the Web perspective as https://github.com/mrlt8/docker-wyze-bridge/issues/867. I do not receive my typical MFA text message and the Web UI does not load.

I fell back to 2.3.0 and it is functional.

Here are my logs:

```date	stream	content
2023/06/29 06:26:53	stdout	[WyzeBridge] 172.17.0.1 - - [29/Jun/2023 10:26:53] "[35m[1mGET / HTTP/1.1[0m" 500 -
2023/06/29 06:26:53	stdout	AssertionError: {'ts': 1688034413633, 'code': '1001', 'msg': 'ParameterError', 'data': {}}
2023/06/29 06:26:53	stdout	           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023/06/29 06:26:53	stdout	    assert str(resp_json.get("code", 0)) == "1", resp_json
2023/06/29 06:26:53	stdout	  File "/app/wyzecam/api.py", line 317, in validate_resp
2023/06/29 06:26:53	stdout	                ^^^^^^^^^^^^^^^^^^^
2023/06/29 06:26:53	stdout	    resp_json = validate_resp(resp)
2023/06/29 06:26:53	stdout	  File "/app/wyzecam/api.py", line 196, in get_homepage_object_list
2023/06/29 06:26:53	stdout	           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023/06/29 06:26:53	stdout	    data = get_homepage_object_list(auth_info)
2023/06/29 06:26:53	stdout	  File "/app/wyzecam/api.py", line 203, in get_camera_list
2023/06/29 06:26:53	stdout	                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023/06/29 06:26:53	stdout	    self.cameras = wyzecam.get_camera_list(self.auth)
2023/06/29 06:26:53	stdout	  File "/app/wyzebridge/wyze_api.py", line 166, in get_cameras
2023/06/29 06:26:53	stdout	           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023/06/29 06:26:53	stdout	    return func(self, *args, **kwargs)
2023/06/29 06:26:53	stdout	  File "/app/wyzebridge/wyze_api.py", line 62, in wrapper
2023/06/29 06:26:53	stdout	             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023/06/29 06:26:53	stdout	    result = func(self, *args, **kwargs)
2023/06/29 06:26:53	stdout	  File "/app/wyzebridge/wyze_api.py", line 46, in wrapper
2023/06/29 06:26:53	stdout	                                      ^^^^^^^^^^^^^^^^^^
2023/06/29 06:26:53	stdout	    return 0 if self.mfa_req else len(self.get_cameras() or [])
2023/06/29 06:26:53	stdout	  File "/app/wyzebridge/wyze_api.py", line 124, in total_cams
2023/06/29 06:26:53	stdout	                                         ^^^^^^^^^^^^^^^^^
2023/06/29 06:26:53	stdout	    cam_data=web_ui.all_cams(wb.streams, wb.api.total_cams, host),
2023/06/29 06:26:53	stdout	  File "/app/frontend.py", line 84, in index
2023/06/29 06:26:53	stdout	           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023/06/29 06:26:53	stdout	    return self.ensure_sync(f)(*args, **kwargs)
2023/06/29 06:26:53	stdout	  File "/usr/local/lib/python3.11/site-packages/flask_httpauth.py", line 174, in decorated
2023/06/29 06:26:53	stdout	           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023/06/29 06:26:53	stdout	    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
2023/06/29 06:26:53	stdout	  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1469, in dispatch_request
2023/06/29 06:26:53	stdout	         ^^^^^^^^^^^^^^^^^^^^^^^
2023/06/29 06:26:53	stdout	    rv = self.dispatch_request()
2023/06/29 06:26:53	stdout	  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1484, in full_dispatch_request
2023/06/29 06:26:53	stdout	         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023/06/29 06:26:53	stdout	    rv = self.handle_user_exception(e)
2023/06/29 06:26:53	stdout	  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1486, in full_dispatch_request
2023/06/29 06:26:53	stdout	               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023/06/29 06:26:53	stdout	    response = self.full_dispatch_request()
2023/06/29 06:26:53	stdout	  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 2190, in wsgi_app
2023/06/29 06:26:53	stdout	Traceback (most recent call last):
2023/06/29 06:26:53	stdout	[2023-06-29 10:26:53,805] ERROR in app: Exception on / [GET]
2023/06/29 06:26:52	stdout	[WyzeBridge] ☁️ Fetching 'cameras' from the Wyze API...
2023/06/29 06:26:52	stdout	[WyzeBridge] 🔍 Could not find local cache for 'cameras'
2023/06/29 06:26:44	stdout	IndexError: list index out of range
2023/06/29 06:26:44	stdout	                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
2023/06/29 06:26:44	stdout	    resp["verification_id"] = creds.mfa_details["totp_apps"][0]["app_id"]
2023/06/29 06:26:44	stdout	  File "/app/wyzebridge/wyze_api.py", line 334, in mfa_response
2023/06/29 06:26:44	stdout	           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023/06/29 06:26:44	stdout	    resp = mfa_response(self.auth, TOKEN_PATH)
2023/06/29 06:26:44	stdout	  File "/app/wyzebridge/wyze_api.py", line 223, in _mfa_auth
2023/06/29 06:26:44	stdout	    self._mfa_auth()
2023/06/29 06:26:44	stdout	  File "/app/wyzebridge/wyze_api.py", line 150, in login
2023/06/29 06:26:44	stdout	             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023/06/29 06:26:44	stdout	    result = func(self, *args, **kwargs)
2023/06/29 06:26:44	stdout	  File "/app/wyzebridge/wyze_api.py", line 46, in wrapper
2023/06/29 06:26:44	stdout	    self.api.login(fresh_data=fresh_data)
2023/06/29 06:26:44	stdout	  File "/app/wyze_bridge.py", line 31, in run
2023/06/29 06:26:44	stdout	    self.run()
2023/06/29 06:26:44	stdout	  File "/usr/local/lib/python3.11/threading.py", line 1038, in _bootstrap_inner
2023/06/29 06:26:44	stdout	Traceback (most recent call last):
2023/06/29 06:26:44	stdout	Exception in thread Thread-1:
2023/06/29 06:26:44	stdout	[WyzeBridge] 🔐 MFA Code Required [Unknown]
2023/06/29 06:26:44	stdout	[WyzeBridge] [33mPress CTRL+C to quit[0m
2023/06/29 06:26:44	stdout	 * Running on http://172.17.0.3:5000
2023/06/29 06:26:44	stdout	 * Running on http://127.0.0.1:5000
2023/06/29 06:26:44	stdout	 * Running on all addresses (0.0.0.0)
2023/06/29 06:26:44	stdout	[WyzeBridge] [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
2023/06/29 06:26:44	stdout	 * Debug mode: off
2023/06/29 06:26:44	stdout	 * Serving Flask app 'frontend'
2023/06/29 06:26:44	stdout	[WyzeBridge] ☁️ Fetching 'auth' from the Wyze API...
2023/06/29 06:26:44	stdout	[WyzeBridge] 🔍 Could not find local cache for 'auth'
2023/06/29 06:26:44	stdout	
2023/06/29 06:26:44	stdout	🚀 DOCKER-WYZE-BRIDGE v2.3.6 
2023/06/29 06:26:44	stdout	```

**Comment by mrlt8** [comment]
Thanks! 

My bad, I thought we could use the "primary_option" from the API, but it turns out that they just return "Unknown" if it's not set so the bridge would attempt to use the time-based one-time password. 

Should be fixed in v2.3.7!

**Comment by Dot50Cal** [comment]
Hmm I am now getting this error with the 2.3.7 version:

```
date	stream	content
2023/06/29 16:26:50	stdout	[WyzeBridge] 👋 goodbye!
2023/06/29 16:26:50	stdout	[WyzeBridge] Stopping 0 streams
2023/06/29 16:26:35	stdout	[WyzeBridge] {"description":"Internal Error","requestId":"74c59122-18a8-49df-b00c-e1b2e8f26293","errorCode":5000}
2023/06/29 16:26:35	stdout	[WyzeBridge] ⚠️ 400 Client Error: Bad Request for url: https://auth-prod.api.wyze.com/v3/user/login
2023/06/29 16:26:34	stdout	[WyzeBridge] ☁️ Fetching 'auth' from the Wyze API...
2023/06/29 16:26:34	stdout	[WyzeBridge] 🔍 Could not find local cache for 'auth'
2023/06/29 16:26:34	stdout	[WyzeBridge] ☁️ Fetching 'cameras' from the Wyze API...
2023/06/29 16:26:34	stdout	[WyzeBridge] 🔍 Could not find local cache for 'cameras'
2023/06/29 16:26:29	stdout	[WyzeBridge] 172.17.0.1 - - [29/Jun/2023 20:26:29] "GET /static/site.js HTTP/1.1" 200 -
2023/06/29 16:26:29	stdout	[WyzeBridge] 172.17.0.1 - - [29/Jun/2023 20:26:29] "GET /static/site.css HTTP/1.1" 200 -
2023/06/29 16:26:29	stdout	[WyzeBridge] 172.17.0.1 - - [29/Jun/2023 20:26:29] "GET / HTTP/1.1" 200 -
2023/06/29 16:26:19	stdout	[WyzeBridge] {"description":"Internal Error","requestId":"0904581e-d12c-4779-988c-b5c16ad0bf6b","errorCode":5000}
2023/06/29 16:26:19	stdout	[WyzeBridge] ⚠️ 400 Client Error: Bad Request for url: https://auth-prod.api.wyze.com/v3/user/login
2023/06/29 16:26:19	stdout	[WyzeBridge] ☁️ Fetching 'auth' from the Wyze API...
2023/06/29 16:26:19	stdout	[WyzeBridge] 🔍 Could not find local cache for 'auth'
2023/06/29 16:26:19	stdout	[WyzeBridge] ☁️ Fetching 'user' from the Wyze API...
2023/06/29 16:26:19	stdout	[WyzeBridge] 🔍 Could not find local cache for 'user'
2023/06/29 16:26:14	stdout	[WyzeBridge] {"description":"Internal Error","requestId":"c8a700ed-b6c1-49e9-90d9-1031ac5b0cda","errorCode":5000}
2023/06/29 16:26:14	stdout	[WyzeBridge] ⚠️ 400 Client Error: Bad Request for url: https://auth-prod.api.wyze.com/v3/user/login
2023/06/29 16:26:13	stdout	[WyzeBridge] ☁️ Fetching 'auth' from the Wyze API...
2023/06/29 16:26:13	stdout	[WyzeBridge] 🔍 Could not find local cache for 'auth'
2023/06/29 16:26:13	stdout	[WyzeBridge] ☁️ Fetching 'cameras' from the Wyze API...
2023/06/29 16:26:13	stdout	[WyzeBridge] 🔍 Could not find local cache for 'cameras'
2023/06/29 16:26:04	stdout	[WyzeBridge] {"description":"Internal Error","requestId":"ec2e74b4-aadb-4eac-a69b-fe5642925e4d","errorCode":5000}
2023/06/29 16:26:04	stdout	[WyzeBridge] ⚠️ 400 Client Error: Bad Request for url: https://auth-prod.api.wyze.com/v3/user/login
2023/06/29 16:26:03	stdout	[WyzeBridge] [33mPress CTRL+C to quit[0m
2023/06/29 16:26:03	stdout	 * Running on http://172.17.0.3:5000
2023/06/29 16:26:03	stdout	 * Running on http://127.0.0.1:5000
2023/06/29 16:26:03	stdout	 * Running on all addresses (0.0.0.0)
2023/06/29 16:26:03	stdout	[WyzeBridge] [31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
2023/06/29 16:26:03	stdout	 * Debug mode: off
2023/06/29 16:26:03	stdout	 * Serving Flask app 'frontend'
2023/06/29 16:26:03	stdout	[WyzeBridge] ☁️ Fetching 'auth' from the Wyze API...
2023/06/29 16:26:03	stdout	[WyzeBridge] 🔍 Could not find local cache for 'auth'
2023/06/29 16:26:03	stdout	
2023/06/29 16:26:03	stdout	🚀 DOCKER-WYZE-BRIDGE v2.3.7 
2023/06/29 16:26:03	stdout	
```

**Comment by mrlt8** [comment]
hmm. Not sure about `400 Client Error: Bad Request`, but I pushed some updates that should hopefully help.

**Comment by Dot50Cal** [comment]
Hey thank you! Whatever happened I believe it is fixed now in 2.3.8. Up & running without issue now, I will close this out.

**Comment by Dot50Cal** [comment]
Hey thank you! Whatever happened I believe it is fixed now in 2.3.8. Up & running without issue now, I will close this out.
